src/profile-logic/process-profile.js

```diff
@@ -382,12 +385,12 @@ function _extractJsFunction(
 ): IndexIntoFuncTable | null {
   // Check for a JS location string.
   const jsMatch: RegExpResult =
-    // Given:   "functionName (http://script.url/:1234)"
-    // Captures: 1^^^^^^^^^^  2^^^^^^^^^^^^^^^^^^ 3^^^
-    /^(.*) \((.*):([0-9]+)\)$/.exec(locationString) ||
-    // Given:   "http://script.url/:1234"
-    // Captures: 2^^^^^^^^^^^^^^^^^ 3^^^
-    /^()(.*):([0-9]+)$/.exec(locationString);
+    // Given:   "functionName (http://script.url/:1234:1234)"
+    // Captures: 1^^^^^^^^^^  2^^^^^^^^^^^^^^^^^^ 3^^^ 4^^^
+    /^(.*) \((.*):([0-9]+):([0-9]+)\)$/.exec(locationString) ||
+    // Given:   "http://script.url/:1234:1234"
+    // Captures: 2^^^^^^^^^^^^^^^^^ 3^^^ 4^^^
+    /^()(.*):([0-9]+):([0-9]+)$/.exec(locationString);
```

Comment:
The column number is optional, profiles from older firefox won't have it. So you need to mark it as optional in the regular expression, and check for it above.

To make it optional you can replace `:([0-9]+)` by `(?::([0-9]+))?` in both expressions (`(?: ... )` is a non-capturing grouping expression, and the final `?` make it optional).

Reply:
@julienw  when i make it optional the column number saved as line number and line number pasted in the file name which is the original issue but when i remove the `?` it works perfectly